datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -119,7 +119,7 @@ public AbstractKafkaConnector(String connectorName, Properties config, GroupIdCo
   protected abstract AbstractKafkaBasedConnectorTask createKafkaBasedConnectorTask(DatastreamTask task);
 
   @Override
-  public void onAssignmentChange(List<DatastreamTask> tasks) {
+  public synchronized void onAssignmentChange(List<DatastreamTask> tasks) {
```

Comment:
do all connectors need this to be synchronized?

Reply:
Technically, it depends on connector implement this, a connector can choose to implement a fine grained logic themselves. I also see other connectors, like a fileConnectopr implement synchronized on the entire method though.

Reply:
Cool! Might want to consider making it a part of the documentation.